ompi/mca/common/ompio/common_ompio_file_read.c

```diff
@@ -352,53 +348,95 @@ int mca_common_ompio_file_read_at (ompio_file_t *fh,
     return ret;
 }
 
+static void mca_common_ompio_post_next_read_subreq(struct mca_ompio_request_t *req, int index)
+{
+    uint32_t iov_count = 1;
+    size_t pos = 0, spc = 0, tbw = 0;
+    int i = 0;
+    mca_ompio_request_t *ompio_subreq=NULL;
+    size_t bytes_per_cycle = OMPIO_MCA_GET(req->req_fh, pipeline_buffer_size);
+    struct iovec decoded_iov;
+
+    /* Step 1: finish index-1 unpack operation */
+    if (index - 1 >= 0) {
+        size_t num_bytes = bytes_per_cycle;
+        /**
+         *  should really be 'req_num_subreqs -1 == index -1'
+         *  which is the same as below.
+         */
+        if (req->req_num_subreqs == index) {
+            num_bytes = req->req_max_data - (index-1)* bytes_per_cycle;
+        }
+        decoded_iov.iov_base = req->req_tbuf;
+        decoded_iov.iov_len  = num_bytes;
+        opal_convertor_unpack (&req->req_convertor, &decoded_iov, &iov_count, &pos);
```

Comment:
This usage of the convertor being linear, it seems to indicate an assumption that requests complete in order. However, I don't see any code that would enforce that. Can you clarify and add a comment about this case.

Reply:
@bosilca thank you for the review. Basically, only one chunk is initiated at a time, and when that chunk finishes it launches the next chunk (implemented in the ompio_progress function). So there is no other way than to execute linearly.

Reply:
That makes sense, and clear some of my other comments as well (the reuse of the convertor as an example). thanks for the clarification.